### PR TITLE
Concordance history

### PR DESCRIPTION
**Pull request by kmarsyd**
Marius to review:

Added a Demo that describes how to use for different data inputs.

Note: Some code redundancy that could be cleaned.

### CODE REVIEW

**Comment by marius-mather** [comment]
There's some automatic formatting set up for this project with [pre-commit](https://pre-commit.com/). Can you run it against all files by:

* Installing `pre-commit` (should already be done in the default Poetry environment)
* Initializing it with `pre-commit install`
* Running against all files: `pre-commit run --all-files`
* Committing the results

**Comment by marius-mather** [comment]
My suggestions for improved readability in `_get_html2()`:

```python
        cell_hover = {'selector': 'td:hover', 'props': [('background-color', '#ffffb3')]}
        index_names = {'selector': '.index_name', 'props': 'font-style: italic; color: darkgrey; font-weight:normal;'}
        headers = {'selector': 'th:not(.index_name)', 'props': 'background-color: #000066; color: white;'}
        align = {'selector': 'td', 'props': 'font-weight: bold;'}
        centre = {'selector': 'th.col_heading', 'props': 'text-align: center;'}
        odd_rows = {"selector": "tbody tr:nth-child(odd)", "props": "background-color: #f9f9f9"}
        even_rows = {"selector": "tbody tr:nth-child(even)", "props": "background-color: #dddddd"}
        column_styles = {
            "match": [{"selector": "td",
                       "props": [("background-color", "#99CC99"), ("text-align", "center"), ("padding-left", "0px"),
                                 ("padding-right", "0px")]}],
            "left_context": [{"selector": "td",
                              "props": [("text-align", "right"), ("padding-right", "0px")]}],
            "right_context": [{"selector": "td",
                               "props": [("text-align", "left"), ("padding-left", "0px")]}],
        }
        html = (
            results.style
            .set_table_styles(column_styles)
            .set_table_styles([cell_hover, index_names, headers, align, centre, odd_rows, even_rows], overwrite=False)
            .to_html()
        )
```

Basically, striped rows, making sure the context columns are not padded where they meet the match column, and trying to set the styles at the column level where possible as I think this should be a bit more efficient (the syntax for this is a little clunky).

File: src/atap_widgets/concordance.py [hunk 1]
```diff
@@ -506,3 +796,162 @@ def get_reversed_words(s: pd.Series):
...
+            style={"description_width": "initial"},
+        )
+        page_input = ipywidgets.BoundedIntText(
+            value=1,
+            min=1,
+            max=1,
+            step=1,
+            description="Page:",
+        )
+        window_width_input = ipywidgets.BoundedIntText(
```
Comment (marius-mather) [inline src/atap_widgets/concordance.py:871]: Can this be set to a default width of 50 and a step size of 1 now? It makes for a much nicer looking default.

File: src/atap_widgets/concordance.py [hunk 2]
```diff
@@ -134,10 +143,147 @@ def prepare_text_df(
 
     if isinstance(language_model, str):
         language_model = spacy.load(language_model)
-    output["spacy_doc"] = output["text"].map(language_model)
+    output["spacy_doc"] = output["text"].map(language_model) #Doc for each line
+    
     return output
 
+class DataIngest():
```
Comment (marius-mather) [inline src/atap_widgets/concordance.py:150]: I think ideally names for functions/classes should mention what they achieve - maybe `ConcordanceLoader` would be a better name for this?

File: src/atap_widgets/concordance.py [hunk 3]
```diff
@@ -134,10 +143,147 @@ def prepare_text_df(
...
+        return df
+
+    def group_by_chunk(self,df):
+        #add row number tags to original df and chunk
+        df['row'] = df.index
+        df['text'] = df['row'].astype(str).str.cat(df['text'],sep = "--") 
+        grouped = df.groupby(['chunk'])['text'].apply(''.join).reset_index()
+        return grouped
+
+    def appify(self,language_model: Union[str, spacy.language.Language] = "en_core_web_sm"):
```
Comment (marius-mather) [inline src/atap_widgets/concordance.py:266]: What does `appify()` mean here? Can we just call this `show()`?

**Review by marius-mather — COMMENTED** [review]
My suggestions for a few small things that will hopefully improve the user experience

File: src/atap_widgets/concordance.py (hunk 1 above)
Comment (kmarsyd) [reply src/atap_widgets/concordance.py:883]: done

File: src/atap_widgets/concordance.py (hunk 2 above)
Comment (kmarsyd) [reply src/atap_widgets/concordance.py]: done

File: src/atap_widgets/concordance.py (hunk 3 above)
Comment (kmarsyd) [reply src/atap_widgets/concordance.py]: done

**Comment by marius-mather** [comment]
Thanks @kmarsyd , merged now, it should be automatically released as v0.5.0 of the package

**Pull request merged**